elastic/integrations
# Add Filestream integration

## Proposed commit message

This commit adds the Filestream integration

## Checklist


## Author's Checklist
 - [x] Use fingerprint as default file identity

## How to test this PR locally

1. Build the integration
2. Start a stack with Elastic-Package or upload the integration to an existing stack
3. Go to the "Add integrations page"
4. Make sure "Display beta integrations" is selected
5. Add the "Custom Filestream integration", configure `paths` to read from any text file
6. Deploy the Elastic-Agent enrolled in this policy
7. Ensure the data from the file is ingested.

The Custom Filestream integration uses the fingerprint file identity by default, so make sure the file used for testing has a size of at least `1024` bytes or change the fingerprint length in the integration settings.

## Related issues
- Closes https://github.com/elastic/elastic-agent/issues/5370

## Screenshots
![2024-11-06_20-33](https://github.com/user-attachments/assets/2048bcd4-1cc6-4aba-ba9f-f6bb46f15c19)
